src/state/l2_book/error.rs

```diff
@@ -0,0 +1,89 @@
+//! Error types for L2/L3 order book operations.
+
+use std::fmt;
+
+use fastnum::UD64;
+
+use super::order::OrderSlot;
+use crate::types::{OrderId, OrderSide};
+
+/// Error type for L2/L3 order book operations.
+#[derive(Debug, Clone, PartialEq)]
+pub enum L2BookError {
```

Comment:
```suggestion
pub enum OrderBookError {
```
As it's both L2 and L3 now, `L2*` naming becomes misleading for shared things